scanner/decisions/0009-nvd-cvss-scores.md

```diff
@@ -0,0 +1,121 @@
+# 0008 - NVD CVSS Scores
+
+- **Author(s):** Ross Tannenbaum
+- **Created:** [2024-04-11 Thurs]
+
+## Status
+
+Status: Accepted.
+
+## Context
+
+Users who require FedRAMP compliance [must be able see NVD's CVSS scores associated with each CVE](https://www.fedramp.gov/assets/resources/documents/CSP_Vulnerability_Scanning_Requirements.pdf).
+
+Currently, Scanner V4 provides a single CVSS score and severity, which is preferably from the vendor.
+In reality, the CVSS scores that are shown are from NVD except in the following cases:
+
+* RHEL-base images
+  * Red Hat-provided CVSS scores/severities are used.
+* OSV-provided data
+  * If OSV.dev has a CVSS score, then that score is used.
+    * The score is converted into a severity based on the CVSS version's specification.
+  * Note: it is likely, if not always true, the score displayed by OSV is the same as NVD.
+
+The current API looks like the following:
+
+```
+message VulnerabilityReport {
+  message Vulnerability {
+    enum Severity {
+      SEVERITY_UNSPECIFIED = 0;
+      SEVERITY_LOW = 1;
+      SEVERITY_MODERATE = 2;
+      SEVERITY_IMPORTANT = 3;
+      SEVERITY_CRITICAL = 4;
+    }
+    message CVSS {
+      message V2 {
+        float base_score = 1;
+        string vector = 2;
+      }
+      message V3 {
+        float base_score = 1;
+        string vector = 2;
+      }
+      V2 v2 = 1;
+      V3 v3 = 2;
+    }
+    ...
+    string severity = 6;
+    ...
+    CVSS cvss = 12;
+  }
+  ...
+}
+```
+
+## Decision
+
+The VulnerabilityReport API will be extended to support more than one CVSS message. This will be done as follows:
+
+```
+message VulnerabilityReport {
+  message Vulnerability {
+    enum Severity {
+      SEVERITY_UNSPECIFIED = 0;
+      SEVERITY_LOW = 1;
+      SEVERITY_MODERATE = 2;
+      SEVERITY_IMPORTANT = 3;
+      SEVERITY_CRITICAL = 4;
+    }
+    message CVSS {
+      message V2 {
+        float base_score = 1;
+        string vector = 2;
+      }
+      message V3 {
+        float base_score = 1;
+        string vector = 2;
+      }
+      V2 v2 = 1;
+      V3 v3 = 2;
+    }
+    ...
+    string severity = 6;
+    ...
+    CVSS cvss = 12;
+    CVSS nvd_cvss = 13; <-- New field.
+  }
+  ...
+}
+```
+
+Currently, there is only a need to have CVSS scores from the vendor (if available) and NVD (if available), no more.
+Making this explicit, then, makes it easy and clear to users to find each scores and choose between the two.
+
+Another option is to provide a more generic solution. One way to do this would be to update `CVSS cvss = 12` to 
+`repeated CVSS cvss = 12` (note: the [proto3 spec](https://protobuf.dev/programming-guides/proto3/#updating) indicates this is compatible).
```

Comment:
Collapsing multiple sources into one could hinder extensibility in the future. Specifically, schema changes are not isolated to one _kind_ of cvss since the struct is being reused.

Reply:
I wonder if we would prefer Scanner to make no decision for Central and return all information it can find with enough metadata so Central can decide how to use the data. For example, returning a list of CVSS scores related to the vuln (Advisory or CVE), with a field representing the "security source" used for the data (this would likely be translatable to an updater or enricher inside Scanner).

This would reduce the coupling between the two services, which means it would facilitate extensibility. Both services could probably change their implementation without bothering each other.